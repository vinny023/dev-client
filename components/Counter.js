import React from 'react';
// import { Layout, State, Text } from '@ui-kitten/components';
import { StyleSheet, View, Text, Button } from 'react-native';
import {connect} from 'react-redux'
import {incrementAction} from '../redux/actions'

class Counter extends React.Component {

    render() {
        return (
            <View style={{flex: 1, flexDirection:'column'}}>
            <Text>counter: {this.props.counter}</Text>
            <Button
            title="+1"
            onPress={this.props.increment}
            />
            </View>
        ) 
    }
}

const mapStateToProps = state => {return {counter: state.counterState.counter}}
const mapDispatchToProps = dispatch => {return {increment: () => dispatch(incrementAction())}}

export default connect(mapStateToProps, mapDispatchToProps)(Counter)