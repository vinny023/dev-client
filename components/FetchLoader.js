import React from 'react'
import {connect} from 'react-redux'
import {Text, View, Button} from 'react-native'
import { fetchAction } from '../redux/actions'



export class FetchLoader extends React.Component {

    constructor(props) {
        super(props)
    }

    render() {  
        const {data, loading, error} = this.props.fetchState
        
        return (
            <View>
            { loading ? <Text>Click Button To Load</Text> : <Text>Loading</Text> }
            { (data !== null) && <Text>{data.rates.aed.name}</Text>}
            <Button
                title="Click Me to Fetch"
                onPress={() => this.props.fetch()}
            />
            </View>

        )
    }
}

const mapDispatchToProps = dispatch => {
    return ({
        fetch: () => dispatch(fetchAction())
    })
}

const mapStateToProps = state => {
    return({
        fetchState: state.fetchState
    })
}

export default connect(mapStateToProps, mapDispatchToProps)(FetchLoader)