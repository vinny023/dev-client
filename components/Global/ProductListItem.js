import React from 'react';
import { StyleSheet, View, Text, Button, TouchableOpacity } from 'react-native';
import * as actions from '../../redux/actions'
import {connect} from 'react-redux'
import _ from 'lodash';

// export class ProductListItem extends React.Component {

//     constructor(props) {
//         super(props)
//    }

const updateQuantity = (masterCart, product) => {
    for (const supplierCart of masterCart) {
        for (const cartItem of supplierCart.cart) {
                if (cartItem.sku === product.sku) {
                   return cartItem.quantity
                }           
        }
    }
    return 0
}

class ProductListItem extends React.PureComponent  {
    
    constructor(props) {
        super(props)
        this.state = {quantity: 0}

        // this.addItem = this.addItem.bind(this)
        // this.subtractItem = this.subtractItem.bind(this)
    }

    componentDidMount() {  

        this.setState({
            quantity: updateQuantity(this.props.masterCart, this.props.item)
        })
    } 
    
    componentDidUpdate(prevProps, prevState) {

        if (!_.isEqual(prevProps.masterCart , this.props.masterCart)) {
            this.setState({
                quantity: updateQuantity(this.props.masterCart, this.props.item)
            })   
        }
    }

    addItem = (payload) => {
        this.props.addItem({item:{...this.props.item}, amount: 1})
    }
    
    subtractItem = (payload) => {
        this.props.subtractItem({item:{...this.props.item}, amount: 1})
    }

    removeItem = () => {
        this.props.subtractItem({item:{...this.props.item}, amount: this.state.quantity})
    }

    render() {  
        
        const {item} = this.props

        //  console.log('PRODUCT LIST ITEM RENDERING')
        // console.log(item)
        // console.log(item.displayName)

    
        return (   
            <View>{
                (!this.props.hideZero || this.state.quantity !== 0) &&
            <View style={{flex: 1, flexDirection: 'row', alignItems:'center', justifyContent:'space-evenly'}}
            key={item.sku}
            >
            <Text>{item.displayName}</Text>
            <Text>{item.price}</Text>
            {
                (item.quantity >= 1) && (<Button 
                    title="Remove"
                    onPress={this.removeItem}
                />)
            }            
            <Button 
                title="-" //MINUS BUTTON 
                onPress = {this.subtractItem}
            />
            <Text>{this.state.quantity}</Text>
            <Button 
                title=" + " //ADD BUTTON
                onPress = {this.addItem}
                />
            </View>
        }
            </View>         
         
            )

        }
    
}


const mapDispatchToProps = dispatch => {
    return (
        {
            addItem:addItemProps => dispatch(actions.addItem(addItemProps)),
            subtractItem:subtractItemProps => dispatch(actions.subtractItem(subtractItemProps))
        }
    )
}

const mapStateToProps = state => {
    return (
        {
            masterCart: state.cartState.masterCart
        }
    )
}

export default connect(mapStateToProps,mapDispatchToProps)(ProductListItem)

