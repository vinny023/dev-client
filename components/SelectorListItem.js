import React from 'react'
import {Text, TouchableOpacity } from 'react-native'
import { useNavigation } from '@react-navigation/native';


export default function SelectorListItem({ screen, item }) {
    const navigation = useNavigation();

    if (screen==="OrderDetailScreen") {
        const order = item
        return (
            <TouchableOpacity 
                style={{padding: 20, backgroundColor: '#FDD7E4', alignSelf: 'stretch',  textAlign: 'center',}}
                onPress={() => navigation.navigate(screen, order)}
            >
            <Text>{order.supplier} - {order.estimatedDeliveryDate}</Text>            
            </TouchableOpacity>
         )
    } 
    
    
}