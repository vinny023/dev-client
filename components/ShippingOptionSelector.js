import React from 'react'
import {Text, View, Image} from 'react-native'
import {Picker} from '@react-native-picker/picker'
import * as actions from '../redux/actions'
import {connect} from 'react-redux'

export class ShippingOptionSelector extends React.Component {
    
    constructor(props) {
        super(props)
        this.state = {
            selectedIndex: 0
        }
    }

    render() {

        //check selected value from cart
        let selectedIndex = 0
        this.props.masterCart.forEach(supplierCart => {
            if (supplierCart.supplier.supplierId === this.props.supplierId) {
                if (supplierCart.shippingOption) {
                    this.props.options.forEach((shippingOption, index) => {
                        if (shippingOption.title === supplierCart.shippingOption.title) {
                            selectedIndex = index;
                        }
                    })
                }
            } 
        })

        console.log(this.props.options)

        return (
            <View>
                <Text>Shipping Options</Text>
                <Picker        
                selectedValue={selectedIndex}        
                style={{ height: 50, width: 150 }}
                onValueChange={(itemValue, itemIndex) => this.props.setShippingOption({supplierId: this.props.supplierId, option: this.props.options[itemIndex]})}
                >
                    {this.props.options.map((option, index) => {
                        return <Picker.Item  key={index} label={option.title} value={index} />
                    })                               
                }
                </Picker>
            </View>
        )
    }    
}

const mapStateToProps = state => {
    return (
        {
            masterCart: state.cartState.masterCart
        }
    )
}

const mapDispatchToProps = dispatch => {
    return (
        {
            setShippingOption: params => dispatch(actions.setShippingOption(params))            
        }
    )
}


export default connect(mapStateToProps, mapDispatchToProps)(ShippingOptionSelector)