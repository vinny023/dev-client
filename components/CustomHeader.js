import React from 'react'
import { View, Text } from 'react-native'
import { useSelector } from 'react-redux'
import { commonStyles } from '../theme'
import { CartButton } from './Global/CartButton'

export default function CustomHeader({title}) {
    const masterCart = useSelector(state => state.masterCart)
    return (
        <View style={[commonStyles.row,{justifyContent:'space-between'}]}>
            <Text>{title}</Text>
            <CartButton/>
        </View>
    )
}
