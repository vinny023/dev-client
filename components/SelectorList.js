import React from 'react'

import SelectorListItem from './SelectorListItem'

export default function SelectorList( {navigation, screen, list, type}) {

    return (
        list.map(listItem => {
          return  <SelectorListItem navigation={navigation} screen={screen} item={listItem}/>
        })    
    )

}