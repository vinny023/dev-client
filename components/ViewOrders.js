import React from 'react'
import {View} from 'react-native'
import { connect } from "react-redux";
import SelectorList from './SelectorList'

export function ViewOrders({ orders }) {

    //any grouping or filtering action
    //title, search ,etc. 

    return(
        <View>
            <SelectorList screen={"OrderDetailScreen"} list={orders}/>
        </View>
    )
}

const mapStateToProps = state => {
    return (
        {
            orders: state.orderState.orders
        }
    )
}

export default connect(mapStateToProps,null)(ViewOrders)
