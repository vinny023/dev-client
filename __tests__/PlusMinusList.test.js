import React from 'react';
import { render, shallow, mount } from 'enzyme';
import {Text, View} from 'react-native'
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-15';
import {create} from 'react-test-renderer'
// import configureStore from 'redux-mock-store'; // Smart components

// Component to be tested
import { PlusMinusList } from '../components/PlusMinusList'
import { ProductListItem } from '../components/ProductListItem'


import {testItemList, testCart} from './setup/TestConfig'

describe('PlusMinus', () => {

    const wrapper = shallow(<PlusMinusList list={testItemList} cart={[]}/>)
    wrapper.debug()
    
    test('renders the component', () => {
        expect(toJson(wrapper)).toMatchSnapshot()
    })

    test('renders the component with ALL elements of list prop with empty cart', () => {
        const listItems = wrapper.find(ProductListItem)
        expect(listItems.length-1).toEqual(testItemList.length)
    })

    test('renders correct cart quantities', () => {
        const wrapper = shallow(<PlusMinusList list={testItemList} cart={testCart}/>)
        const listItems = wrapper.find(View)

        const check = [];
        testCart.map(cartItem => {
            itemViewWrapper = wrapper.findWhere(node => node.key() === cartItem.sku)
            if (itemViewWrapper.length > 0) {
              cartItemWrapper = wrapper.findWhere(node => node.key() === cartItem.sku).first()          
              check.push(cartItemWrapper.contains(<Text>{cartItem.quantity}</Text>))
            } 
        })

        expect(check).toEqual(testCart.map(() => true))        
    })      

 })