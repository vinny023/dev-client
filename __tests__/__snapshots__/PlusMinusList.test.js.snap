// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusMinus renders the component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "start",
    }
  }
>
  <Connect(ProductListItem)
    item={
      Object {
        "name": "Avocados",
        "price": 2000,
        "sku": "avoc123",
      }
    }
    key="0"
    style="verticalList"
  />
  <Connect(ProductListItem)
    item={
      Object {
        "name": "Chiasdfsdcken",
        "price": 2000,
        "sku": "chic123",
      }
    }
    key="1"
    style="verticalList"
  />
  <Connect(ProductListItem)
    item={
      Object {
        "name": "Pork",
        "price": 2000,
        "sku": "pork123",
      }
    }
    key="2"
    style="verticalList"
  />
  <Connect(ProductListItem)
    item={
      Object {
        "name": "Bananas",
        "price": 2000,
        "sku": "banan123",
      }
    }
    key="3"
    style="verticalList"
  />
  <Connect(ProductListItem)
    item={
      Object {
        "name": "Hummus",
        "price": 2000,
        "sku": "humm123",
      }
    }
    key="4"
    style="verticalList"
  />
  <Connect(ProductListItem)
    item={
      Object {
        "name": "Chips",
        "price": 2000,
        "sku": "chip123",
      }
    }
    key="5"
    style="verticalList"
  />
  <Connect(ProductListItem)
    item={
      Object {
        "name": "Paper Plates (100 ct)",
        "price": 2000,
        "sku": "pap100123",
      }
    }
    key="6"
    style="verticalList"
  />
  <Connect(ProductListItem)
    item={
      Object {
        "name": "Packaging",
        "price": 2000,
        "sku": "pack123",
      }
    }
    key="7"
    style="verticalList"
  />
</View>
`;
